aws/container-linux/kubernetes/controllers.tf

```diff
@@ -1,3 +1,16 @@
+locals {
+  user_data = {
+    ignition = {
+      version = "2.3.0"
+      config = {
+        replace = {
+          source = "s3://${aws_s3_bucket.ignition_configs.id}/${element(aws_s3_bucket_object.controller-ignitions.*.id, count.index)}"
```

Comment:
You can't use `count.index` in this context, this must be inside `aws_instance.controllers`. If passing a map inline to `jsonencode` doesn't work then I'd just revert to the JSON string you had before.